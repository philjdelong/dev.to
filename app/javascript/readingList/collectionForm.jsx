import { h, Component } from 'preact';
import PropTypes from 'prop-types';
import linkState from 'linkstate';
import Tags from '../shared/components/tags';

export class CollectionForm extends Component {
  constructor({ tagList }) {
    super({ tagList });
    this.state = {
      // eslint-disable-next-line react/no-unused-state
      title: '',
      tagList,
      // auth: false
    };
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  submitCollection = (e, title, tags) => {
    // check that tag length is > 1, title req handled on input
    // need to modify route also and ensure error handling is included.
    e.preventDefault();
    fetch('/reading_collections', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'X-CSRF-Token': window.csrfToken,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        title,
        tags,
      }),
      credentials: 'same-origin',
    });
  };

  render() {
    const { tagList } = this.state;

    return (
      <div className="articleformcontainer">
        <section className="articleform">
          <form className="articleform__form">
            <h1>Add your new collection</h1>
            <input
              type="text"
              name="title"
              placeholder="Title..."
              onChange={this.handleChange}
              className="articleform__title"
              required
            />
            <Tags
              defaultValue={tagList}
              onInput={linkState(this, 'tagList')}
              maxTags={4}
              autoComplete="off"
              classPrefix="articleform"
            />
            {/* <h4>{`${this.state.error}`}</h4> */}
            <a
              className="articleform__buttons--publish collection-btn"
              type="button"
              href="readinglist"
            >
              CREATE COLLECTION
            </a>
          </form>
        </section>
      </div>
    );
  }
}

CollectionForm.propTypes = {
  tagList: PropTypes.arrayOf(PropTypes.object).isRequired,
};
