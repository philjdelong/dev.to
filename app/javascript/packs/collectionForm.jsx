import { h, render } from 'preact';
import { getUserDataAndCsrfToken } from '../chat/util';
import { CollectionForm } from '../readingList/collectionForm';

function loadElement() {
  getUserDataAndCsrfToken().then(() => {
    const root = document.getElementById('collection-list');
    if (root) {
      render(<CollectionForm tagList="" />, root, root.firstElementChild);
    }
  });
}

window.InstantClick.on('change', () => {
  loadElement();
});

loadElement();
