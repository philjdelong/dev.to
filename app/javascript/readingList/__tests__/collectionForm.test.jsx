import { h } from 'preact';
import render from 'preact-render-to-json';
import { CollectionForm } from '../collectionForm';

describe('<CollectionForm />', () => {
  it('renders properly', () => {
    const tree = render(<CollectionForm tagList="" />);
    expect(tree).toMatchSnapshot();
  });
});
