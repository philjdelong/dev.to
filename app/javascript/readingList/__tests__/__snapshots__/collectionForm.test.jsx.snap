// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollectionForm /> renders properly 1`] = `
<section
  class="articleform"
>
  <form
    class="articleform__form"
  >
    <h1>
      Add your new collection
    </h1>
    <input
      class="articleform__title"
      name="title"
      onChange={[Function]}
      placeholder="Collection Name..."
      required={true}
      type="text"
    />
    <div
      class="articleform__tagswrapper"
    >
      <input
        autoComplete="off"
        class="articleform__tags"
        id="tag-input"
        onBlur={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        onKeyDown={[Function]}
        placeholder="4 tags max, comma separated, no spaces or special characters"
        ref={[Function]}
        type="text"
        value=""
      />
    </div>
    <button
      class="articleform__buttons--publish button collection-btn"
      onClick={[Function]}
      type="button"
    >
      CREATE COLLECTION
    </button>
  </form>
</section>
`;
